test/new-e2e/utils/e2e/stack_definition.go

```diff
@@ -83,6 +85,10 @@ func AgentStackDef(vmParams []ec2params.Option, agentParameters ...agentparams.O
 				return nil, err
 			}
 
+			if pipelineId, ok := os.LookupEnv("CI_PIPELINE_ID"); ok {
```

Comment:
Pipeline ID is an internal data that should not be exposed to the user. It should be injected through the config map. See [this](https://github.com/DataDog/datadog-agent/blob/7.46.0/test/new-e2e/runner/configmap.go#L74-L106).